samcli/lib/providers/sam_stack_provider.py

```diff
@@ -218,6 +219,8 @@ def get_stacks(
             that might want to get substituted within the template and its child templates
         metadata: Optional[Dict]
             Optional dictionary of nested stack resource metadata values.
+        template_dict_format: Optional[Dict]
+            Optional dictionary representing the sam template file to be used instead of the template file
```

Comment:
I was a bit confused by this before I looked into the implementation. I think just `template_dict` might be a better name, and saying `representing the sam template` rather than `representing the sam template file`. Also should leave some message that one of this one (and only one) of either this one or template file is required.

Reply:
Also, since now the template file is not required (as they could provide the dict) we should probable reflect that in the function's parameters and make it an Optional type. Would also need to update the comment to reflect this (and probably also include again here a line that one and only one of either this or template_dict is required). Also, in that case, should handle below in the code what happens if the user doesn't provide template file or dict (should throw an exception)

Reply:
Made the fixes and added the exception raise for if neither are passed